fiftyone/utils/hf_hub.py

```diff
@@ -0,0 +1,1109 @@
+from concurrent.futures import ThreadPoolExecutor
```

Comment:
Please add our standard module docstring to the top. Something like this:

```
"""
Utilities for working with `Hugging Face <https://huggingface.co>`_.

| Copyright 2017-2024, Voxel51, Inc.
| `voxel51.com <https://voxel51.com/>`_
|
"""
```

Reply:
As an aside, how would you feel about renaming the module to `fiftyone.utils.huggingface`? I'm thinking that interacting with HF Hub is a pretty core aspect of how FiftyOne and HF integrate, so it would be nice for the workflows to have "prime" branding:

```py
import fiftyone.utils.huggingface as fouh

dataset = fouh.load_from_hub(...)
fouh.push_to_hub(...)
```

Noting of course that the Transformers stuff has its own `fiftyone.utils.transformers` module.

In truth we probably should have started nesting things one level further `fiftyone.utils.huggingface.transformers` and `fiftyone.utils.huggingface.hf_hub` and then imported the important stuff in `fiftyone/utils/huggingface/__init__.py`, but that ship has already sailed for `transformers`.

If we use `fiftyone.utils.huggingface` for this PR, we could move stuff to a submodule in the future and import the public functions one-level higher in a backwards-compatible way, if we decided that this module is getting too big.

Reply:
Yeah I'm with you. I ran into this when doing the first iteration of docs for the hub integration, realizing a lot of headers potentially needed to be changed. Moving to this paradigm for the future.